examples/jsm/controls/OrbitControls.js

```diff
@@ -101,6 +101,13 @@ class OrbitControls extends Controls {
 		this.keyPanSpeed = 7.0;	// pixels moved per arrow key push
 		this.zoomToCursor = false;
 
+		// Set constraints to how far you can pan
+		this.panLimits = {
+			x: { min: null, max: null },
+			y: { min: null, max: null },
+			z: { min: null, max: null },
+		};
```

Comment:
Tip: In three.js, we would typically use `Box3` instead, along with `Box3.clampPoint( point, target )`.